vyxal/elements.py

```diff
@@ -135,26 +134,15 @@ def all_diagonals(lhs, ctx):
     Diagonals of a matrix, starting with the main diagonal.
     """
 
-    @lazylist
-    def gen():
-        vector = list(map(lambda x: iterable(x, ctx=ctx), lhs))
-        diag_num = 0
-        current_diagonal = numpy.diag(vector)
-        # postive diags first
-        while len(current_diagonal):
-            yield vyxalify(current_diagonal)
-            diag_num += 1
-            current_diagonal = numpy.diag(vector, k=diag_num)
-
-        diag_num = -1
-        current_diagonal = numpy.diag(vector, k=diag_num)
-        # now the other diagonals
-        while len(current_diagonal):
-            yield vyxalify(current_diagonal)
-            diag_num -= 1
-            current_diagonal = numpy.diag(vector, k=diag_num)
-
-    return gen()
+    vector = [iterable(x, ctx=ctx) for x in lhs]
+    all_diags = [[] for _ in range(len(vector) * 2 - 1)]
+    start = 0
+    print(vector)
+    for row in vector:
```

Comment:
This code looks like it relies on the matrix to be square...

Reply:
It does, why?

Reply:
If that's the spec, never mind.